src/node_dotenv.cc

```diff
@@ -105,15 +105,22 @@ Local<Object> Dotenv::ToObject(Environment* env) const {
   return result;
 }
 
+// Removes space characters (spaces, tabs and newlines) from
+// the start and end of a given input string
 std::string_view trim_spaces(std::string_view input) {
   if (input.empty()) return "";
-  if (input.front() == ' ') {
-    input.remove_prefix(input.find_first_not_of(' '));
+
+  auto pos_start = input.find_first_not_of(" \t\n");
+  if (pos_start == std::string_view::npos) {
+    return "";
   }
-  if (!input.empty() && input.back() == ' ') {
-    input = input.substr(0, input.find_last_not_of(' ') + 1);
+
+  size_t pos_end = input.find_last_not_of(" \t\n");
+  if (pos_end == std::string_view::npos) {
+    pos_end = input.size() - 1;
   }
-  return input;
+
+  return input.substr(pos_start, pos_end - pos_start + 1);
```

Comment:
Please also check for `pos_start == pos_end` before returning substring

Reply:
check added 🙂